From: Guoqing Jiang <gqjiang@suse.com>
Date: Mon, 31 Oct 2016 10:19:00 +0800
Subject: md/bitmap: call bitmap_file_unmap once bitmap_storage_alloc returns
 -ENOMEM
References: bsc#1003941,FATE#321732
Git-commit: cbb387323610295be9d36c51287b668c1140704f
Patch-Mainline: v4.10-rc1

It is possible that bitmap_storage_alloc could return -ENOMEM,
and some member inside store could be allocated such as filemap.

To avoid memory leak, we need to call bitmap_file_unmap to free
those members in the bitmap_resize.

Reviewed-by: NeilBrown <neilb@suse.com>
Signed-off-by: Guoqing Jiang <gqjiang@suse.com>
Signed-off-by: Shaohua Li <shli@fb.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/bitmap.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/md/bitmap.c b/drivers/md/bitmap.c
index 09a6310..a826b4c 100644
--- a/drivers/md/bitmap.c
+++ b/drivers/md/bitmap.c
@@ -2056,8 +2056,10 @@ int bitmap_resize(struct bitmap *bitmap, sector_t blocks,
 					   !bitmap->mddev->bitmap_info.external,
 					   mddev_is_clustered(bitmap->mddev)
 					   ? bitmap->cluster_slot : 0);
-	if (ret)
+	if (ret) {
+		bitmap_file_unmap(&store);
 		goto err;
+	}
 
 	pages = DIV_ROUND_UP(chunks, PAGE_COUNTER_RATIO);
 
-- 
1.8.5.6

